src/json-symtab-language/json_symbol_table.h

```diff
@@ -0,0 +1,17 @@
+/*******************************************************************\
+
+Module: JSON symbol table deserialization
+
+Author: Chris Smowton, chris.smowton@diffblue.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_JSON_SYMTAB_LANGUAGE_JSON_SYMBOL_TABLE_H
+#define CPROVER_JSON_SYMTAB_LANGUAGE_JSON_SYMBOL_TABLE_H
+
+class jsont;
+class symbol_tablet;
+
+void symbol_table_from_json(const jsont &, symbol_tablet &);
+
```

Comment:
We now have rvalues! Return that symbol_table!

Reply:
Do you suggest to manually construct and populate the symbol table from within the function? That does make sense, but it's not very convenient, because the symbol table is constructed much earlier in the call stack, and it's in the interface of `languaget::typecheck`, and passed around to end up in this function that populates it.

Reply:
the symbol table that gets passed to json_symtab_languaget::typecheck is expected to be empty; so you can simply overwrite.